kasa/smart/smartdevice.py

```diff
@@ -47,7 +47,8 @@ def __init__(
         self._components_raw: dict[str, Any] | None = None
         self._components: dict[str, int] = {}
         self._state_information: dict[str, Any] = {}
-        self.modules: dict[str, SmartModule] = {}
+        self._modules: dict[str, SmartModule] = {}
+        self._combined_modules: dict[str, SmartModule] | None = None
```

Comment:
Why not just use the `self._modules` for storing the initialized modules?

Reply:
self._modules is used for the `update`.  At the moment we have decided not to call update on the child modules and if we add the child modules to the parents `_modules` I think it could be confusing down the road.  I think it's better to keep the combined separate from `_modules` to make it clearer what's going on.

Reply:
Did you test it it'd cause issues otherwise? I'd rather keep the update logic cleaner, considering this concerns now only a single device.